pkg/config/spec.go

```diff
@@ -324,11 +334,18 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 		}
 	}
 
-	// Deep look to see if upgrade.recovery == true in the config
-	// if yes, we set the upgrade spec "Entry" to "recovery"
+	// One way to set the entry is to use the cli arg "--recovery"
+	// which makes config.upgrade.entry be "recovery"
+	// Another way is by setting the cli arg "boot-entry" which set it to the
+	// specified value.
+	// Lastly, user can set "upgrade.recovery: true" in the kairos config, which
+	// should result in entry being "recovery".
 	entry := ""
 	_, ok := cfg.Config["upgrade"]
 	if ok {
+		// check value from --recovery and --boot-entry
+		entry, _ = cfg.Config["upgrade"].(collector.Config)["entry"].(string)
```

Comment:
We were skipping this but not tests were broken. This means we need to add a test to make sure we don't break it again.

Reply:
It seems that this wasn't needed because of the unmarshallFullSpec below: https://github.com/kairos-io/kairos-agent/pull/499/files#diff-53b724197bc14c3828ce6725322b892cb788860b0526184b63d7d114edb60941R370

`generateUpgradeConfForCLIArgs` has added the field `upgrade.entry` (see `ExtraConfigUpgrade`) which is marshalled into a yaml and passed to the config collector (as a string).

Let's check if this is still working as intended.